### ISSUE TITLE
How to get a trained model instance by its name  in Tabular task?

### ISSUE THREAD

**Issue by razmikmelikbekyan** [awslabs/autogluon#432]
What is the most suitable way to get the trained model instance including all data preprocessing steps in Tabular prediction? 

Image you have trained autogluon tabular task and now you want to use one of the autogluon models as a separate model instance. The idea is the following. 

```
import autogluon as ag
from autogluon.task import TabularPrediction as task
import pandas as pd

df_train = pd.read_csv('AutogluonModels/titanic/train.csv')

output_directory = 'AutogluonModels/my_models'
predictor = task.fit(
    train_data=df_train,
    holdout_frac=None, 
    label='Survived',
    output_directory=output_directory,
    time_limits=60,
    auto_stack=False,
    stack_ensemble_levels=1
)
```

So now we have a predictor object. It would be nice to have some `get_model(model_name)` method which will return the specified model including its data preprocessing steps. So you can use only this model for refit or something else. This question arise because I wanted to use only the ensemble model from autogluon for cross validation. So I had to refit all the models.

**Comment by Innixma** [comment]
Could you explain the last part, specifically: 

> This question arise because I wanted to use only the ensemble model from autogluon for cross validation. So I had to refit all the models.

What do you mean for cross-validation, are you running AutoGluon 5 times to get a CV score?  Would the ability to return OOF (Out of Fold) predictions solve your issue? This is added in #431. Could you provide pseudo-code that achieves what you require?

To get the model object at present, you would do:

`model = predictor._trainer.load_model('my_model')`

To get X into the form required for inference (Does not work for the training set):
```
X_raw = task.Dataset(...)
X_preprocessed = predictor._learner.transform_features(X_raw)
X = predictor._trainer.get_inputs_to_model(model='my_model', X=X_preprocessed, preprocess=False)
y_pred_proba = model.pred_proba(X)
```

However, this y_pred_proba is still the internal pred proba representation, so you would need further transformation code to make it return equivalent to `predictor.pred_proba(X_raw, model='my_model')`

This will get significantly simpler after #431 is merged.

**Comment by Innixma** [comment]
Note also that the model object does not know all the context required to transform the raw data into its internal representation. To return a model with that knowledge, it would be to effectively return a copy of the Learner object which has all other models removed from it.

**Comment by razmikmelikbekyan** [comment]
> Note also that the model object does not know all the context required to transform the raw data into its internal representation, nor to convert its output to the format seen by the user. To return a model with that knowledge, it would be to effectively return a copy of the Learner object which has all other models removed from it.

Sorry for late reply. 
Below you can see the pseudo code: 
```cv = KFold(random_state=45, shuffle=True)
cv_scores = []
for train_ind, valid_ind in cv.split(X_train):
    ag_train_data = X_train.iloc[train_ind].copy() 
    ag_train_data['target'] = y_train.iloc[train_ind].tolist()
    
    ag_valid_data = X_train.iloc[valid_ind]
    ag_valid_target = y_train.iloc[valid_ind]
    
    regressor = task.fit(
        train_data=ag_train_data,
        label='target',
        time_limits=30,
        eval_metric='mean_absolute_error',
        stopping_metric='mean_absolute_error', 
        auto_stack=True,
        verbosity=0,
    )
    valid_pred = regressor.predict(ag_valid_data, model='weighted_ensemble_k0_l1')
    cv_scores.append(mean_absolute_error(ag_valid_target, valid_pred))
```
 Yep I now that there are 2 levels of data preprocessing: one is model agnostic and the second one is done by each model. So taking into consideration this, this suggestion (copying learner) I think is the most suitable.

**Comment by Innixma** [comment]
Thanks for the reply, it was very helpful! I understand now what you are going for.

Interestingly, when auto_stack=True, we can actually bypass the for-loop here. This is an alternative strategy to get your cv_scores:

```
X_train['target'] = y_train

predictor = task.fit(
        train_data=X_train,
        label='target',
        time_limits=30,
        auto_stack=True,
        verbosity=0,
)

model = 'ExtraTreesClassifierEntr_STACKER_l0'
predictor_info = predictor._learner.get_info(include_model_info=True)
model_info = predictor_info['model_info'][model]
cv_scores = [child_info['val_score'] for child, child_info in model_info['children_info'].items()]
```

Note that this won't work for weighted ensembles, since they don't perform bagging. Implementing cv_score retrieval for weighted ensemble would be a bit tricky, since it never truly performs cv.

> So you can use only this model for refit or something else. This question arise because I wanted to use only the ensemble model from autogluon for cross validation. So I had to refit all the models.

Can you provide a pseudo code example that shows what you mean and showcases the proposed new functionality?
